### ISSUE TITLE
`shorthand_operator` should not trigger for forwarding operator implementations.

### ISSUE THREAD

**Issue by lhunath** [realm/SwiftLint#4611]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

When implementing our own shorthand operators, it is common to implement it by delegating to non-shorthand operators, eg.
```swift
    public static func -= (lhs: inout Self, rhs: Int) {
        lhs = lhs - rhs
    }
```

This pattern should not trigger the linter, since the suggestion is non-operative (it would break the code by introducing a recursive loop), and the pattern is sufficiently common that a general exclusion is warranted.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
error: Shorthand Operator Violation: Prefer shorthand operators (+=, -=, *=, /=) over doing the operation and assigning. (shorthand_operator)
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? `0.49.1`
* Installation method used (Homebrew, CocoaPods, building from source, etc)? `brew`
* Which Xcode version are you using (check `xcodebuild -version`)? `14.1 14B47b`

**Comment by marcelofabri** [comment]
Thanks for the great idea! Implemented in https://github.com/realm/SwiftLint/pull/4613
